### ISSUE TITLE
Pipe function shouldn't inside internal package

### ISSUE THREAD

**Issue by Himani-relan** [nhooyr/websocket#340]
Pipe function here: https://github.com/nhooyr/websocket/blob/master/internal/test/wstest/pipe.go#L17 is inside the internal package which means it can't be imported by other projects.
However it should be external so that it can be imported by other projects. It can support unit testing of the projects.

**Comment by nhooyr** [comment]
I figured you'd always want to use `httptest` instead to setup two real websockets. But I suppose this can be released under a `wstest` package.

**Comment by leosunmo** [comment]
I found this internal package useful for testing. I have a struct that stays around for the duration of a websocket connection, and it has a `*websocket.Conn` field. For testing I don't know how to use `httptest` effectively to set up a dummy connection (if what I'm testing doesn't actually need to send anything over the Conn).

Some advice on how I could use standard `httptest` to (succinctly) create a `*websocket.Conn`, or any comment on potentially exporting this package would be appreciated!

**Comment by mafredri** [comment]
@leosunmo thanks for sharing your experience.

A fairly simple example of using `httptest` can be found in the echo example over here: https://github.com/coder/websocket/blob/e4379472fe1dfe70032ecc68fec08b1b3a8fc996/internal/examples/echo/server_test.go#L18

I'll think about how we can help guide users in writing tests better and we will also consider exporting the `wstest` package. This project has _very_ limited use of `wstest.Pipe`, however, so I'd love to hear more about use-cases where it has been useful!

**Comment by leosunmo** [comment]
Yeah that works really well when I am testing the client part, but less well when I am testing the server part. For example, I want to test how well my server handles a client not responding to the Close handshake. I do something like this:
```
r := manualWSReq(t, "")
w := httptest.NewRecorder()
buf := bytes.NewBuffer(nil)

conn, err := websocket.Accept(w, r, nil)
if err != nil {
	t.Fatalf("failed to accept ws req: %s", err)
}
```
which leads to 
```failed to accept ws req: failed to accept WebSocket connection: http.ResponseWriter does not implement http.Hijacker```

This is presumable the same issue as https://github.com/coder/websocket/issues/455.

**Comment by encbladexp** [comment]
I would also prefer having the `internal` test stuff available for testing my own application.
